src/syscheckd/create_db.c

```diff
@@ -312,51 +312,63 @@ int fim_file(char *file, fim_element *item, whodata_evt *w_evt, int report) {
 }
 
 
-void fim_realtime_event(char *file, fim_element *item) {
+void fim_realtime_event(char *file) {
 
     struct stat file_stat;
 
     // If the file exists, generate add or modify events.
     if (w_stat(file, &file_stat) >= 0) {
 
 #ifdef WIN32
+        fim_element *item;
```

Comment:
Allocating heap memory for a structure like this may be too complex. Please consider allocating this object in the stack:

```c
fim_element item = { .mode = FIM_REALTIME };
fim_checker(file, &item, NULL, 1);
```